From 1ceefd0a85e568caec2007c61910d049c2d14da0 Mon Sep 17 00:00:00 2001
From: Nicolas Fella <nicolas.fella@gmx.de>
Date: Fri, 19 Dec 2025 10:52:23 +0100
Subject: [PATCH] Revert "Use KirigamiAddons to initialise the app"

KirigamiApp changes the QSurfaceFormat, which breaks
the GStreamer integration

BUG: 513544

This reverts commit 56b652b9b84a498ad712680423e45400edf0a43e.

(cherry picked from commit 27a464754ef15992781920c3524945aabd601dd3)
---
 .kde-ci.yml        | 1 -
 CMakeLists.txt     | 3 +--
 src/CMakeLists.txt | 3 +--
 src/main.cpp       | 5 +----
 4 files changed, 3 insertions(+), 9 deletions(-)

diff --git a/.kde-ci.yml b/.kde-ci.yml
index 0d32a6a..80a2336 100644
--- a/.kde-ci.yml
+++ b/.kde-ci.yml
@@ -11,7 +11,6 @@ Dependencies:
     'frameworks/ki18n': '@stable-kf6'
     'frameworks/purpose': '@stable-kf6'
     'frameworks/knotifications': '@stable-kf6'
-    'libraries/kirigami-addons': '@latest-kf6'
 
 Options:
  require-passing-tests-on: ['@all']
diff --git a/CMakeLists.txt b/CMakeLists.txt
index e975f7d..026dcf6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -48,11 +48,10 @@ find_package(PkgConfig REQUIRED)
 pkg_check_modules(GStreamer IMPORTED_TARGET gstreamer-1.0>=${GSTREAMER_MIN_VERSION})
 pkg_check_modules(GStreamerVideo IMPORTED_TARGET gstreamer-video-1.0)
 
-include(ECMFindQmlModule)
+include(ECMQMLModules)
 
 ecm_find_qmlmodule(org.kde.purpose 1.0)
 ecm_find_qmlmodule(org.kde.kirigami 1.0)
-find_package(KF6KirigamiAddons 1.10.0 REQUIRED)
 
 add_definitions(-DQT_NO_URL_CAST_FROM_STRING)
 
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index f875067..d3e3985 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -17,11 +17,10 @@ add_executable(kamoso
 )
 
 kconfig_add_kcfg_files(kamoso kamosoSettings.kcfgc GENERATE_MOC)
-target_link_libraries(kamoso PRIVATE
+target_link_libraries(kamoso
     Qt::Core Qt::Gui Qt::Widgets Qt::Quick
     KF6::KIOFileWidgets KF6::ConfigGui KF6::I18n KF6::Notifications
     PkgConfig::GStreamerVideo PkgConfig::GStreamer
-    KirigamiApp
 )
 
 install(TARGETS kamoso ${KDE_INSTALL_TARGETS_DEFAULT_ARGS})
diff --git a/src/main.cpp b/src/main.cpp
index 8efc42c..35fc481 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -11,16 +11,13 @@
 #include "video/webcamcontrol.h"
 #include <QApplication>
 #include <QIcon>
-#include <KirigamiApp>
 
 #include "kamoso_version.h"
 #include <gst/gst.h>
 
 int main(int argc, char *argv[])
 {
-    KirigamiApp::App app(argc, argv);
-    KirigamiApp kapp;
-
+    QApplication app(argc, argv);
     KLocalizedString::setApplicationDomain("kamoso");
     {
         KAboutData about(QStringLiteral("kamoso"), i18n("Kamoso"), QStringLiteral(KAMOSO_VERSION_STRING), i18n("Utility for taking photos and videos using a webcam"),
-- 
2.52.0

